### PR TITLE
Compute new brace-pair tags when the view is scrolled

### PR DESCRIPTION
**Pull request by CyrusNajmabadi**

### CODE REVIEW

File: src/EditorFeatures/Core/BracePairs/BracePairsTaggerProvider.cs [hunk 1]
```diff
@@ -53,7 +53,11 @@ internal sealed class BracePairsTaggerProvider : AsynchronousViewTaggerProvider<
 
         protected override ITaggerEventSource CreateEventSource(ITextView? textView, ITextBuffer subjectBuffer)
         {
+            this.ThreadingContext.ThrowIfNotOnUIThread();
+            Contract.ThrowIfNull(textView);
```
Comment (sharwell) [inline src/EditorFeatures/Core/BracePairs/BracePairsTaggerProvider.cs:57]: 😕 Why? Nothing here uses the UI thread.

**Review by sharwell — CHANGES_REQUESTED** [review]

File: src/EditorFeatures/Core/BracePairs/BracePairsTaggerProvider.cs (hunk 1 above)
Comment (CyrusNajmabadi) [reply src/EditorFeatures/Core/BracePairs/BracePairsTaggerProvider.cs:57]: This just is so it can fire if we ever break this contract.

Comment (sharwell) [reply src/EditorFeatures/Core/BracePairs/BracePairsTaggerProvider.cs:57]: Would prefer to remove both of these. The first can be removed as both unnecessary and undesirable (only the `ITaggerEventSource.Connect` should be using the main thread), and the second can be removed per https://github.com/dotnet/roslyn/pull/67454#discussion_r1146427239.

[the remark this answers is not in the document]
Comment (CyrusNajmabadi) [reply src/EditorFeatures/Core/BracePairs/BracePairsTaggerProvider.cs]: done.  good idea :)

Comment (CyrusNajmabadi) [reply src/EditorFeatures/Core/BracePairs/BracePairsTaggerProvider.cs]: removed.

**Review by sharwell — APPROVED** [review]

**Pull request merged**